app/models/repo.rb

```diff
@@ -92,9 +92,9 @@ def populate_issues!
     background_populate_issues(self.id)
   end
 
-  def self.exists_with_name?(name)
+  def self.exists_with_name?(name, repos)
     user_name, repo_name = name.downcase.split(?/)
-    Repo.exists?(user_name: user_name, name: repo_name)
+    repos.map { |r| r.full_name }.include?("#{user_name}/#{repo_name}")
```

Comment:
It's doing one query for each of the user's repos, as opposed to one query total.